### ISSUE TITLE
indent const complexe structure

### ISSUE THREAD

**Issue by ZiperRom1** [eslint/eslint#6671]
Hi,

I like all things to be align, and fortunately my awesome IDE PhpStorm 2016 has a quite nice configurable auto-format coding style.
I managed to make it work with all my eslint rules (with a eslint:all + some overrides) expect a complex one, here it is:

PhpStorm 2016 align

``` js
const bowerDirectory = path.join('.', 'bower_components'),
      paths          = {
          'sass'   : {
              'src' : path.join('.', 'resources/assets/sass/app.scss'),
              'dest': path.join('.', 'public')
          },
          'js'     : {
              'src' : `${path.join('.', 'public/js')}/**/*.js`,
              'dest': path.join('.', 'public')
          },
          'vendors': {
              'js'  : {
                  'src' : [
                      path.join(bowerDirectory, 'bootstrap/dist/js/bootstrap.js'),
                      path.join(bowerDirectory, 'domReady/domReady.js'),
                      path.join(bowerDirectory, 'jquery/dist/jquery.js'),
                      path.join(bowerDirectory, 'lodash/dist/lodash.js'),
                      path.join(bowerDirectory, 'requirejs/require.js')
                  ],
                  'dest': path.join('.', 'public/js/vendors')
              },
              'sass': {
                  'src' : [
                      `${path.join(bowerDirectory, 'bootstrap/scss')}/**/*.scss`,
                      `!${path.join(bowerDirectory, 'bootstrap/scss/_variables.scss')}`
                  ],
                  'dest': path.join('.', 'resources/assets/sass/vendors')
              }
          }
      };
```

eslint align (with `eslint --fix` command)

``` js
const bowerDirectory = path.join('.', 'bower_components'),
            paths    = {
                'sass': {
                    'src' : path.join('.', 'resources/assets/sass/app.scss'),
                    'dest': path.join('.', 'public')
                },
                'js': {
                    'src' : `${path.join('.', 'public/js')}/**/*.js`,
                    'dest': path.join('.', 'public')
                },
                'vendors': {
                    'js': {
                        'src': [
                            path.join(bowerDirectory, 'bootstrap/dist/js/bootstrap.js'),
                            path.join(bowerDirectory, 'domReady/domReady.js'),
                            path.join(bowerDirectory, 'jquery/dist/jquery.js'),
                            path.join(bowerDirectory, 'lodash/dist/lodash.js'),
                            path.join(bowerDirectory, 'requirejs/require.js')
                        ],
                        'dest': path.join('.', 'public/js/vendors')
                    },
                    'sass': {
                        'src': [
                            `${path.join(bowerDirectory, 'bootstrap/scss')}/**/*.scss`,
                            `!${path.join(bowerDirectory, 'bootstrap/scss/_variables.scss')}`
                        ],
                        'dest': path.join('.', 'resources/assets/sass/vendors')
                    }
                }
            };
```

So basically there is two problems
- eslint do not want object properties to be aligned with a none-4 multiple value (because based on variable declaration which is aligned with a const key-word so it starts with a 6 spaces align value).
- eslint do not want object properties to be aligned on semicolons if there is more than one blank line between object properties even if they are on the same hierarchy level.

Here is the eslint report on PhpStorm 2016 align

```
  13:7   error  Expected indentation of 12 space characters but found 6  indent
  14:11  error  Extra space after key 'sass'                             key-spacing
  18:11  error  Extra space after key 'js'                               key-spacing
  23:15  error  Extra space after key 'js'                               key-spacing
  24:19  error  Extra space after key 'src'                              key-spacing
  34:19  error  Extra space after key 'src'                              key-spacing

✖ 6 problems (6 errors, 0 warnings)
```

eslint version

```
λ eslint -v
v3.0.1
```

eslintrc.json

``` json
{
    "env"          : {
        "browser": true,
        "es6"    : true,
        "node"   : true
    },
    "extends"      : "eslint:all",
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType" : "module"
    },
    "parser"       : "babel-eslint",
    "rules"        : {
        "indent"         : [
            "error",
            4,
            {
                "VariableDeclarator": {
                    "var"  : 2,
                    "let"  : 2,
                    "const": 3
                }
            }
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes"         : [
            "error",
            "single"
        ],
        "semi"           : [
            "error",
            "always"
        ],
        "max-len"        : [
            "error",
            120
        ],
        "padded-blocks"  : [
            "error",
            "never"
        ],
        "func-style"     : [
            "error",
            "declaration"
        ],
        "no-multi-spaces": [
            "error",
            {
                "exceptions": {
                    "Property"          : true,
                    "VariableDeclarator": true,
                    "ImportDeclaration" : true
                }
            }
        ],
        "key-spacing"    : [
            "error",
            {
                "singleLine": {
                    "beforeColon": false,
                    "afterColon" : true
                },
                "multiLine" : {
                    "beforeColon": false,
                    "afterColon" : true,
                    "align"      : "colon"
                }
            }
        ],
        "dot-location"   : [
            "error",
            "property"
        ]
    }
}
```

Sorry for the post length due to code examples :p
Thanks for this awesome tool !

**Comment by vitorbal** [comment]
Hi @ZiperRom1, thank you for your issue! I'm not sure I understand what you are asking, though.
Are you suggesting a new option for those rules to be able to get the same type of alignment that your IDE uses? Or is this a bug report?

**Comment by ZiperRom1** [comment]
Hi, there are two evolutions to solve this.
- Allow `key-spacing` to add spaces on multi-level object properties in the same hierarchy level even if there is more than one line between them.
- Allow variables to be indented with a non 4 multiple value (eg 6) if they follow a `const` declaration.

It is purely stylistic, the way this last rule is done currently is something like
```js
const totoTheBest = 1,
                theBest = 2;
```

While this alignment would be preferable because the object starts more on the left and so you gain some spaces then in the object.
```js
const totoTheBest = 1,
          theBest       = 2;
```

To be more precise

The actual way
```js
const aSuperLongNameBecauseItMayExist = 1,
                                                  myObject = {
                                                      "property1": "value1",
                                                      "property2": "value2"
                                                  };
```

The prefered way
```js
const aSuperLongNameBecauseItMayExist = 1,
         myObject                                         = {
             "property1": "value1",
             "property2": "value2"
         };
```

**Comment by ilyavolodin** [comment]
Basically, you are asking to change the behavior of all `VariableDeclarator` options from multiplier to actual number of spaces require?

**Comment by ZiperRom1** [comment]
Yes, more like an exception properties with a list of static allowed values.
Or any rules / combo that can accept the "prefered way" indentation with indentation rule activated.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
